src/TraceEvent/TraceEvent.cs

```diff
@@ -3079,7 +3079,7 @@ public abstract unsafe class TraceEventDispatcher : TraceEventSource
         /// </summary>
         /// <param name="traceFileName">A path to a trace file.</param>
         /// <returns>A TraceEventDispatcher for the given trace file.</returns>
-        public static TraceEventDispatcher GetDispatcherFromFileName(string traceFileName)
+        public static TraceEventDispatcher GetDispatcherFromFileName(string traceFileName, Etlx.TraceLogOptions options = null)
```

Comment:
@brianrob the API doesn't change, this augments it. Prior to this we didn't pass start offset, I had forgotten to include changes the Bing XAP team had requested. Or I can make this API addition clearer but adding the options as a separate method definition?

Reply:
Understood that it's essentially a different overload.  I'm saying that if we could avoid this, that would be preferable since this API is generic and the values being passed in are really specific to BPerf (at least at this point).  None of the other formats support this functionality.

We also shouldn't mix TraceLog stuff with Dispatchers.  TraceLog is a layer above TraceEventDispatcher.  At a minimum if we are going to make this change we should use a new Dispatcher specific object.  I'd like to think about a way to make this specific to BPerf if possible, but I haven't found one just yet.  How do you envision using this?  I see options getting passed inside TraceLog creation, but it doesn't look like the new values get set anywhere (or did I miss this?)

Reply:
You didn't miss it, it's because XAP will use it in their TraceEvent-based tool. I can work on this to make PerfView use a different path for BTL files and so that mix doesn't happen.

You're right in that those layers are distinct, so mixing it up is not great.

Thanks for the comment will get back to you.

Reply:
Sounds good @mjsabby.  Let me know what you're thinking.  I'll see if I can come up with a good strategy here.  Does XAP need to build TraceLog files or are they going to just create and use the BPerfEventSource?

Reply:
Yeah both XAP and SNR depend on TraceLog heavily. I'm taking my time to ensure this is the last change we need to do.

Reply:
@brianrob @LakshanF still working on this. I'm context switching to removing the PInvoke for decompression in my previous change and introduce the managed version (it's all source code, no packages). I'm working on that because that PInvoke is not supported on all Windows OSes.

Reply:
Sounds good.  Thanks for the update @mjsabby.

Reply:
@mjsabby, how is this coming up?

Reply:
This is ready for a review once again. @brianrob I couldn't find another way to expose these options. The public API is only being augmented, it is not changing the existing surface area.